test/e2e_node/critical_pod_test.go

```diff
@@ -111,6 +124,24 @@ var _ = SIGDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:CriticalPod]
 	})
 })
 
+func createSyncInNamespaceFromPod(f *framework.Framework, pod *v1.Pod) *v1.Pod {
```

Comment:
That was my first try. Let's see the code:
```
// CreateSync creates a new pod according to the framework specifications, and wait for it to start and be running and ready.
func (c *PodClient) CreateSync(pod *v1.Pod) *v1.Pod {
	namespace := c.f.Namespace.Name
	p := c.Create(pod)
	ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(c.f.ClientSet, p.Name, namespace, PodStartTimeout)) # <<<----- NOTE THE NAMESPACE HERE
	// Get the newest pod after it becomes running and ready, some status may change after pod created, such as pod ip.
	p, err := c.Get(context.TODO(), p.Name, metav1.GetOptions{})
	ExpectNoError(err)
	return p
}
```
(https://github.com/kubernetes/kubernetes/blob/v1.21.2/test/e2e/framework/pods.go#L100)
The problem is the wait, which is done in the framework-built namespace, which is possibly different from the one on which the pod is created.

Fixing the framework is probably the best approach, but I'm not completely sure how, and extra care is needed because this function is used in a ton of places

Reply:
oh I see, it created all pods under `pod.ObjectMeta.Namespace = kubeapi.NamespaceSystem`

Reply:
just build your own pod creator, it is 3 lines 
```
func createSyncPodOrFail(f *framework.Framework, pod *v1.Pod) *v1.Pod {
		p, err = f.ClientSet.CoreV1().Pods(ns).Create(context.TODO(), pod,  metav1.CreateOptions{})
		framework.ExpectNoError(err)
		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, pod.Name, f.Namespace.Name, framework.PodStartTimeout))
                return p
```

Reply:
yep, we'll keep these functions I inlined.